base/strings/util.jl

```diff
@@ -359,6 +359,7 @@ _replace(io, repl::Function, str, r, pattern) =
     print(io, repl(SubString(str, first(r), last(r))))
 
 function replace(str::String, pattern, repl, limit::Integer)
+    limit == 0 && return str
```

Comment:
While you're at it, maybe rename the argument so that it's consistent with the docstring (or change the docstring)?